gateway/sds_gateway/users/files_utils.py

```diff
@@ -0,0 +1,358 @@
+"""
+Utility functions extracted from FilesView class.
+
+These functions were originally methods in the FilesView class but have been
+extracted to improve testability and reusability. They can now be tested
+independently without needing to instantiate the entire view class.
+"""
+
+import logging
+
+from django.utils import timezone
+
+from sds_gateway.api_methods.models import Capture
+from sds_gateway.api_methods.models import File
+from sds_gateway.api_methods.models import ItemType
+from sds_gateway.api_methods.models import UserSharePermission
+
+logger = logging.getLogger(__name__)
+
+
+def get_filtered_files_queryset(base_queryset):
+    """
+    Apply common file filtering to exclude system files and hidden files.
+
+    Args:
+        base_queryset: Base File queryset to filter
+
+    Returns:
+        QuerySet: Filtered queryset excluding system files
+    """
+    return (
+        base_queryset.exclude(name__endswith=".DS_Store")
+        .exclude(name__startswith=".")
+        .exclude(name__in=[".DS_Store", "._.DS_Store", "Thumbs.db", "desktop.ini"])
+    )
+
+
+def format_modified(dt: timezone.datetime | None) -> str:
+    """Return a consistent display string for modified timestamps."""
+    return dt.strftime("%Y-%m-%d %H:%M") if dt else "N/A"
+
+
+def make_dir_item(
+    *,
+    name: str,
+    path: str,
+    uuid: str = "",
+    is_capture: bool = False,
+    is_shared: bool = False,
+    is_owner: bool = False,
+    capture_uuid: str = "",
+    modified_at_display: str = "N/A",
+    shared_by: str = "",
+) -> dict:
+    """Create a standardized directory item dict for the template."""
+    return {
+        "type": "directory",
+        "name": name,
+        "path": path,
+        "uuid": uuid,
+        "is_capture": is_capture,
+        "is_shared": is_shared,
+        "is_owner": is_owner,
+        "capture_uuid": capture_uuid,
+        "modified_at": modified_at_display,
+        "shared_by": shared_by,
+    }
```

Comment:
Linking to that pydantic comment: this function (if it still exists) would return an instance of that data class, that doesn't overload the meaning of `path`.